CMakeLists.txt

```diff
@@ -1,479 +1,44 @@
-else()
-	set(DB_INCLUDE_DIRS ${LEVELDB_INCLUDE_DIRS})
-	set(DB_LIBRARIES ${LEVELDB_LIBRARIES})
-endif()
-
-if (EVMJIT)
-	if (NOT DEFINED LLVM_DIR)
-		if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
-			set(LLVM_DIR "${CMAKE_SOURCE_DIR}/extdep/install/windows/x64/share/llvm/cmake")
-		elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-			set(LLVM_DIR "/usr/local/opt/llvm/share/llvm/cmake")
-		endif()
-	endif()
-
-	set(EVMJIT_CPP TRUE) # include CPP-JIT connector
-	add_subdirectory(evmjit)
-	if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
-		set(EVMJIT_DLLS_LOCAL $<TARGET_FILE:evmjit>)
-		set(EVMJIT_DLLS optimized ${EVMJIT_DLLS_LOCAL} debug ${EVMJIT_DLLS_LOCAL})
-	endif()
-endif()
-
-if (TOOLS OR GUI OR TESTS)
-	set(GENERAL 1)
-else ()
-	set(GENERAL 0)
-endif ()
-
-add_subdirectory(libdevcore)
-if (GENERAL)
-	add_subdirectory(libevmcore)
-	add_subdirectory(libevmasm)
-	add_subdirectory(liblll)
-endif ()
-
-if (SERPENT)
-	add_subdirectory(libserpent)
-	add_subdirectory(sc)
-endif ()
-
-if (TOOLS)
-	add_subdirectory(lllc)
-endif ()
-
-if (JSCONSOLE)
-	add_subdirectory(libjsengine)
-	add_subdirectory(libjsconsole)
-endif ()
-
-if (NOT WIN32)
-	add_definitions(-DETH_HAVE_SECP256K1)
-	add_subdirectory(secp256k1)
-endif ()
-
-add_subdirectory(libscrypt)
-add_subdirectory(libdevcrypto)
-add_subdirectory(libnatspec)
-
-if (GENERAL)
-	add_subdirectory(libp2p)
-endif ()
-
-if (GENERAL OR MINER)
-	add_subdirectory(libethash)
-	if (ETHASHCL)
-		add_subdirectory(libethash-cl)
-	endif ()
-endif ()
-
-add_subdirectory(libethcore)
-
-if (GENERAL)
-	add_subdirectory(libevm)
-	add_subdirectory(libethereum)
-endif ()
-
-if (MINER OR TOOLS)
-	add_subdirectory(ethminer)
-endif ()
-
-if (ETHKEY OR TOOLS)
-	add_subdirectory(ethkey)
-endif ()
-
-# TODO: sort out tests so they're not dependent on webthree/web3jsonrpc and reenable
-#if (TESTS)
-	add_subdirectory(libtestutils)
-	add_subdirectory(test)
-#endif ()
+include(EthDependencies)
 
-if (TOOLS)
+include(EthUtils)
 
-	add_subdirectory(rlp)
-	add_subdirectory(abi)
-	add_subdirectory(ethvm)
+include(EthOptions)
+configure_project()
+print_config()
 
-#	if("x${CMAKE_BUILD_TYPE}" STREQUAL "xDebug")
-#		add_subdirectory(exp)
-#	endif ()
+add_subdirectory(utils)
 
-endif()
+add_subdirectory(dev)
+add_subdirectory(ethereum)
+add_subdirectory(webthree)
 
-# Optional sub-projects
-if (EXISTS "${CMAKE_SOURCE_DIR}/solidity/CMakeLists.txt")
+if (SOLIDITY)
 	add_subdirectory(solidity)
 endif()
 
-if (EXISTS "${CMAKE_SOURCE_DIR}/webthree/CMakeLists.txt")
-	add_subdirectory(webthree)
-endif()
-
-# Optionally if the user has dot/grapviz, generate the dependency graph
-find_program(GRAPHVIZDOT dot)
-if (GRAPHVIZDOT)
-	message(STATUS "Graphviz detected. Drawing a module dependency graph at ${CMAKE_SOURCE_DIR}/dependency_graph.svg")
-	execute_process(
-		COMMAND dot -Tsvg dependency_graph.dot -o dependency_graph.svg
-		WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-	)
+if (GUI)
+	add_subdirectory(alethzero)
+	add_subdirectory(mix)
```

Comment:
They are allowed. If the run `cmake -DGUI=0` alethzero and mix submodules can be still uninitalized.